modules/caddyhttp/matchers_test.go

```diff
@@ -545,6 +545,82 @@ func TestHeaderREMatcher(t *testing.T) {
 	}
 }
 
+func TestVarREMatcher(t *testing.T) {
+	for i, tc := range []struct {
+		match      MatchVarRE
+		input      VarsMiddleware
+		expect     bool
+		expectRepl map[string]string
+	}{
+		{
+			match:  MatchVarRE{"Var1": &MatchRegexp{Pattern: "foo"}},
+			input:  VarsMiddleware{"Var1": "foo"},
+			expect: true,
+		},
+		{
+			match:  MatchVarRE{"Var1": &MatchRegexp{Pattern: "$foo^"}},
+			input:  VarsMiddleware{"Var1": "foobar"},
+			expect: false,
+		},
+		{
+			match:      MatchVarRE{"Var1": &MatchRegexp{Pattern: "^foo(.*)$", Name: "name"}},
+			input:      VarsMiddleware{"Var1": "foobar"},
+			expect:     true,
+			expectRepl: map[string]string{"name.1": "bar"},
+		},
+		{
+			match:  MatchVarRE{"{http.request.method}": &MatchRegexp{Pattern: "^G.[tT]$"}},
+			input:  VarsMiddleware{},
+			expect: true,
+		},
+		{
+			match:  MatchVarRE{"{http.vars.Var1}": &MatchRegexp{Pattern: "[vV]ar[0-9]"}},
+			input:  VarsMiddleware{"Var1": "var1Value"},
+			expect: true,
+		},
+	} {
```

Comment:
Though now followed in the other testcases in this file, this test is well positioned to leverage sub tests in go. 

Ref: https://blog.golang.org/subtests

I would suggest adding a `desc` field to your test case structure above, which would help articulate what you're trying to test in each case in addition to meeting the needs for sub tests. It also looks like these could be run in parallel no probs, but I could be missing something

```
tc := tc // capture range value
t.Run(tc.desc, func(t *testing.T) {
    t.Parallel()
    ... contents of for loop ...
}
```